bin/flutter.bat

```diff
@@ -47,6 +47,18 @@ IF NOT EXIST "%flutter_root%\.git" (
 REM Ensure that bin/cache exists.
 IF NOT EXIST "%cache_dir%" MKDIR "%cache_dir%"
 
+REM If the cache still doesn't exist, fail with an error that we probably don't have permissions.
+IF NOT EXIST "%cache_dir%" (
+  ECHO Error: Unable to create cache directory at
+  ECHO            %cache_dir%
+  ECHO.
+  ECHO        This may be because flutter doesn't have write permissions for
+  ECHO        this path. Try moving flutter to a writable folder, such as
```

Comment:
nit: maybe say "try moving the flutter **directory**..." since we don't want people to just move the flutter executable?

Reply:
Done. I've also changed the next instance of `folder` to `location` since it seemed weird to mix directory and folder and if I made them all the same there were three instances of `directory` in that line (move your flutter directory to a writeable directory such as your home directory).

Now reads:

> This may be because flutter doesn't have write permissions for
> this path. Try moving the flutter directory to a writable location,
> such as within your home directory.

I'll land this, but feel free to request changes (or just change it) if you're not sold on it :)